test/src/test/java/com/rbmhtechnology/vind/test/ServerTest.java

```diff
@@ -3851,4 +3852,43 @@ public void testSpellcheckResultHasGroups() {
         assertEquals(2, result.getResults().size());
         assertEquals(2, result.getFacetResults().getTermFacet(created).getValues().size());
     }
+
+    @Test
+    @RunWithBackend({Solr, Elastic})
+    public void testSpellcheckResultHasUsedSearchterm() {
+
+        FieldDescriptor<String> textField = new FieldDescriptorBuilder()
+                .setFullText(true)
+                .setSuggest(true)
+                .setFacet(true)
+                .buildTextField("textField");
+
+
+        DocumentFactory assets = new DocumentFactoryBuilder("asset")
+                .addField(textField)
+                .build();
+
+        Document d1 = assets.createDoc("1")
+                .setValue(textField, "Stark");
+        Document d2 = assets.createDoc("2")
+                .setValue(textField, "Thor");
+        Document d3 = assets.createDoc("3")
+                .setValue(textField, "Widow");
+
+        SearchServer server = testBackend.getSearchServer();
+
+        server.index(d1);
+        server.index(d2);
+        server.index(d3);
+        server.commit();
+
+        SearchResult result = server.execute(
```

Comment:
add here the test that altering the fulltext search that is passed to the executor does not change the search object in the result